### PR TITLE
Add Database migration service template

### PR DESCRIPTION
**Pull request by ash1625**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 101-azure-database-migration-service/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+            "defaultValue": "SourceServer1",
+            "type": "String",
+            "metadata":
+            {
+                "description": "Name to use for the Source Server."
+            }
+        },
+        "createPublicIP":
+        {
+            "defaultValue": "false",
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,638 @@
+
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters":
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,638 @@
...
+    {
+        "DMSServiceName":
+        {
+            "defaultValue": "AzureDataMigrationTemplateService",
+            "type": "String",
+            "metadata":
+            {
+                "description": "Name of the Azure Data Migration service."
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,638 @@
...
+            }
+        },
+        "sourceServerName":
+        {
+            "defaultValue": "SourceServer1",
+            "type": "String",
+            "metadata":
+            {
+                "description": "Name to use for the Source Server."
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,638 @@
...
+            "defaultValue": "false",
+            "type": "bool",
+            "metadata":
+            {
+                "description": "Do you want to create a public IP address for the source server?"
+            }
+        },
+        "sourceWindowsAdminUserName":
+        {
+            "defaultValue": "sourceAdminUserName",
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,638 @@
...
+            "defaultValue": "targetsqldbservername",
+            "type": "String",
+            "metadata":
+            {
+                "description": "Name to use for the Target Azure SQL DB Server."
+            }
+        },
+        "targetSqlDbAdministratorLogin":
+        {
+            "defaultValue": "targetSqlDbAdminUserName",
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,638 @@
...
+                    {
+                        "publisher": "MicrosoftSQLServer",
+                        "offer": "SQL2016SP1-WS2016",
+                        "sku": "Standard",
+                        "version": "latest"
+                    },
+                    "osDisk":
+                    {
+                        "osType": "Windows",
+                        "name": "[concat(parameters('sourceServerName'),'_OsDisk_1_d05cc07cf42346948bcfd0ccd7c86041')]",
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,638 @@
...
+                {
+                    "computerName": "[parameters('sourceServerName')]",
+                    "adminUsername": "[parameters('sourceWindowsAdminUserName')]",
+                    "adminPassword": "[parameters('sourceWindowsAdminPassword')]",
+                    "windowsConfiguration":
+                    {
+                        "provisionVMAgent": true,
+                        "enableAutomaticUpdates": true
+                    },
+                    "secrets": []
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,638 @@
...
+        },
+        {
+            "type": "Microsoft.Network/networkSecurityGroups",
+            "name": "[variables('sourceServerNSG')]",
+            "apiVersion": "2018-02-01",
+            "location": "[resourceGroup().location]",
+            "scale": null,
+            "properties":
+            {
+                "securityRules": [],
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,638 @@
...
+                            "sourceAddressPrefixes": [],
+                            "destinationAddressPrefixes": []
+                        }
+                    }
+                ]
+            },
+            "dependsOn": []
+        },
+        {
+            "condition": "[equals(parameters('createPublicIP'), bool('true'))]",
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,638 @@
...
+            "location": "[resourceGroup().location]",
+            "scale": null,
+            "properties":
+            {
+                "publicIPAddressVersion": "IPv4",
+                "publicIPAllocationMethod": "Dynamic",
+                "idleTimeoutInMinutes": 4,
+                "ipTags": []
+            },
+            "dependsOn": []
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 12]
```diff
@@ -0,0 +1,638 @@
...
+                "subnets": [
+                    {
+                        "name": "default",
+                        "properties":
+                        {
+                            "addressPrefix": "10.2.0.0/24"
+                        }
+                    }
+                ],
+                "virtualNetworkPeerings": [],
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 13]
```diff
@@ -0,0 +1,638 @@
...
+                "enableDdosProtection": false,
+                "enableVmProtection": false
+            },
+            "dependsOn": []
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 14]
```diff
@@ -0,0 +1,638 @@
...
+            "type": "Microsoft.Network/publicIPAddresses",
+            "sku":
+            {
+                "name": "Basic",
+                "tier": "Regional"
+            },
+            "name": "[variables('publicIPSourceServer')]",
+            "apiVersion": "2018-02-01",
+            "location": "[resourceGroup().location]",
+            "scale": null,
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: remove scale?

[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 15]
```diff
@@ -0,0 +1,638 @@
...
+                "ipTags": []
+            },
+            "dependsOn": []
+        },
+        {
+            "type": "Microsoft.Network/virtualNetworks",
+            "name": "[variables('adVNet')]",
+            "apiVersion": "2018-02-01",
+            "location": "[resourceGroup().location]",
+            "scale": null,
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 16]
```diff
@@ -0,0 +1,638 @@
...
+                    "SQLAuthUpdatePassword": "[parameters('sourceSqlAuthenticationPassword')]"
+                }
+            }
+        },
+        {
+            "type": "Microsoft.DataMigration/services/projects",
+            "name": "[concat(parameters('DMSServiceName'), '/', 'SqlToSqlDbMigrationProject')]",
+            "apiVersion": "2017-11-15-privatepreview",
+            "location": "[resourceGroup().location]",
+            "scale": null,
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 17]
```diff
@@ -0,0 +1,638 @@
...
+            },
+            "dependsOn": [
+                "[resourceId('Microsoft.DataMigration/services', parameters('DMSServiceName'))]"
+            ]
+        },
+        {
+            "type": "Microsoft.Network/virtualNetworks/subnets",
+            "name": "[concat(variables('adVNet'), '/', variables('defaultSubnetName'))]",
+            "apiVersion": "2018-02-01",
+            "scale": null,
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

File: 101-azure-database-migration-service/azuredeploy.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Removed the field

File: 101-azure-database-migration-service/azuredeploy.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Removed the field

File: 101-azure-database-migration-service/azuredeploy.json [hunk 18]
```diff
@@ -0,0 +1,638 @@
...
+        "databaseName": "TargetDatabaseName1",
+        "storageKeyType": "SharedAccessKey"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Compute/virtualMachines",
+            "name": "[parameters('sourceServerName')]",
+            "apiVersion": "2017-12-01",
+            "location": "[resourceGroup().location]",
+            "scale": null,
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: done

File: 101-azure-database-migration-service/README.md [hunk 19]
```diff
@@ -0,0 +1,14 @@
+
+# Azure Database Migration Service
+
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/README.md]: Done

File: 101-azure-database-migration-service/azuredeploy.json [hunk 20]
```diff
@@ -0,0 +1,638 @@
...
+                    {
+                        "collation": "SQL_Latin1_General_CP1_CI_AS",
+                        "maxSizeBytes": 2147483648,
+                        "zoneRedundant": "false"
+                    },
+                    "resources": [
+                        {
+                            "name": "Import",
+                            "type": "extensions",
+                            "apiVersion": "2014-04-01-preview",
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Where do I find the latest api version for the resource? Is the api version resource specific?

File: 101-azure-database-migration-service/azuredeploy.json [hunk 21]
```diff
@@ -0,0 +1,638 @@
...
+            "metadata":
+            {
+                "description": "Administrator Password for the Target Azure SQL DB Server."
+            }
+        }
+    },
+    "variables":
+    {
+        "scriptLocation": "./AddDatabaseToSqlServer.ps1 ",
+        "scriptFiles": ["https://datamigrationwebpi.blob.core.windows.net/templatefiles/AddDatabaseToSqlServer.ps1", "https://datamigrationwebpi.blob.core.windows.net/templatefiles/AdventureWorks2016.bak"],
```
[the remark this answers is not in the document]
Comment (ash1625) [reply 101-azure-database-migration-service/azuredeploy.json]: Done

**Comment by bmoore-msft** [comment]
Can you check the merge conflicts?

**Comment by ash1625** [comment]
Hey Brian we can close this pull request, I had some issues with my branch so had to create a new repo. will close this one

**Pull request closed without merging**